lib/Charts/Title.js

```diff
@@ -70,20 +73,20 @@ const Title = {
             // Enter.
             const addedTitleComponents = titleComponents.enter().append('span')
                 .attr('class', 'title-component');
+
             if (titleSettings.type === 'legend') {
                 addedTitleComponents.append('span')
                     .attr('class', 'color');
             }
             addedTitleComponents.append('span')
                 .attr('class', 'name');
             // Enter and update.
-            titleComponents.select('.color').style('background-color', d=>d.color);
-            titleComponents.select('.name').text(getName);
+            addedTitleComponents.select('.color').style('background-color', d=>d.color);
+            addedTitleComponents.select('.name').text(getName);
```

Comment:
`selection` in v4 are immutable,  so after `selection.data()` , the `addedTitleComponents` is the new selection returned, so trying to modify `titleComponents` at the first render does not actually modify the attributes of latest selection: https://github.com/d3/d3/blob/master/CHANGES.md#selections-d3-selection